user_tools/src/spark_rapids_pytools/cloud_api/sp_types.py

```diff
@@ -851,6 +851,17 @@ class ClusterBase(ClusterGetAccessor):
     props: AbstractPropertiesContainer = field(default=None, init=False)
     logger: Logger = field(default=ToolLogging.get_and_setup_logger('rapids.tools.cluster'), init=False)
 
+    @staticmethod
+    def _verify_workers_configurations(has_no_workers_cb: Callable[[], bool]):
```

Comment:
minor: is this function needed as it's not actually checking configuration, only checking if there are workers?  you could just do check inline in __post_init__ to raise RuntimeError().  i'm guessing you prefer checks done in functions and if so, maybe rename this to just be a function named verify_workers_exist().